### ISSUE TITLE
We need more powered furniture

### ISSUE THREAD

**Issue by I-am-Erk** [CleverRaven/Cataclysm-DDA#53647]
### Is your feature request related to a problem? Please describe.

@Fris0uman quietly added one of the most often requested features, stationary power grids, [way back in October](https://github.com/CleverRaven/Cataclysm-DDA/pull/51286). This feature hasn't had as much fanfare and attention as it richly deserves because so far we have no furniture using it!

### Solution you would like.

We need a pretty basic content addition stage here for all existing things that should be part of your base power grid but only exist as vehicle furniture right now.

- Almost anything we have code for as powered vehicle furniture should get an `APPLIANCE` version, such as laundry machines, forges, kilns, etc.
- Most of the `APPLIANCE` versions can, where appropriate, have higher capacity than the vehicle versions
- Some vehicle furniture (eg kiln) should be removed entirely and replaced with stationary forms where appropriate

Additionally, in terms of building these appliances, we should often change the way the non-powered furniture is placed and used. The process here should be:
- a furniture item, such as `f_washing_machine`, should gain the `DECONSTRUCT_EASY` flag, and when deconstructed should yield a new item, `washing_machine`, rather than component parts. This item can then be installed as the new `APPLIANCE` form of a washing machine, representing unplugging it from the wall and hooking it up to your base power grid; or, it can be deconstructed more thoroughly to get the parts we used to be able to get from a washing machine.

Finally, we should create or move some standing lamps and other powered items that don't belong into a base to use this framework as well, which would mean creating some new `APPLIANCE` types that aren't mirroring stuff from vehicles.

**Stretch goal: better deconstruction**
- A keen contributor could, at this point, also dramatically improve the deconstruction of common appliances. I would suggest for example that a washing machine should first deconstruct into a washing machine cabinet, a washing machine door, and a washing machine drum, as well as some mechanicals and plumbing. Then those parts could if desired be turned into sheet metal, but we could also look into having them used in other things. This kind of more granular deconstruction could allow us to do fun things with the component parts of appliances, such as using the box of a fridge to make a smoker.

### Describe alternatives you have considered.

nothing really, we need this. As stated above, a first pass implementation here doesn't have to include a full audit of deconstruction of appliances. That would just be a bonus.

### Additional context
I am working on a demo of how to do all the parts of this in #53648 - I will include more detailed deconstruction and all the other bits to give a full example

This is a 0.G release blocker. We don't have to have everything, but we should get basic powered furniture for the major crafting and cooking stuff at least.

**Comment by Photoloss** [comment]
What is the general plan for "off-grid infrastructure" components which are kinda necessary to match the quality of life of current vehicles? Storage batteries, real-life UPSes, tanks with plumbing, control panels for electronics (or remote controls), charging stations and so on *basically* use the same components in both situations barring voltage and current compatibility. This also raises the question of how to notify players of compatibility (or lack thereof) as "install in vehicle" and "place as appliance" basically are independent properties and neither is made clear anywhere on the item.

> Most of the APPLIANCE versions can, where appropriate, have higher capacity than the vehicle versions

But not from the same item. This also speaks for the "partial disassembly" follow-up as a single kitchen fridge might yield enough insulated box material to build 2-3 minifridges. *Technically* a full-size fridge would fit on certain vehicles, but it would take up the entire space from the floor of the interior compartment up to the roof height of a van or RV. Can we do that with the existing vehicle code, block out multiple "layers" and flag the tile as higher than a car all in a single component?

> Some vehicle furniture (eg kiln) should be removed entirely and replaced with stationary forms where appropriate

What speaks against running the kiln in a vehicle? What it *should* do is output a huge amount of heat while in use thus rendering a closed car/truck too hot to comfortably work in without serious AC. And if we're mentioning the kiln at least that one stays shut during use, the forge needs to be opened several times and glowing metal swung around your workshop (which might still fit into a large RV in a pinch).
A more interesting case is the washing machine, spun up it's an "involuntary gyroscopic stabiliser" and will put severe strain on its mounting points when you turn a corner. While [certainly amusing](https://youtu.be/vROdVsU_K80?t=66) that's a reason not to put it in a car if I've ever seen one!

There's also the question of what happens to the regular item versions of certain "large tools". Would [this](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/items/tool/metalworking.json#L142) become the "unplugged" APPLIANCE item or would the battery-powered version continue to exist separately including the option of using it within a vehicle?

**Comment by I-am-Erk** [comment]
A kiln as it presently exists isn't vehicle furniture. Vehicle kilns and, to a lesser extent, forges should require features we do not currently support, such as heat use, levelling, and multi-tile vehicle parts.

Washing machines are another one that could be moved to another item.

I'm not sure what you're talking about with minifridges and stuff. Appliances are handled differently, they're only vehicle parts on the code side. You can't install an appliance fridge in your car, so there's no need for any further special flags or anything.

The remainder is kind of beyond this issue: I just want simple additions of appliance versions of the easy stuff, to start. Storage batteries, tanks, etc belong in #53646 discussion.

**Comment by Photoloss** [comment]
> multi-tile vehicle parts

[This](https://potterycrafters.com/wp-content/uploads/2020/03/potter-using-kiln.jpg) looks like it'd fit where a car seat was torn out. Not boardable of course but it sounds like I'm misunderstanding what this "electric kiln" is supposed to represent.

> You can't install an appliance fridge in your car, so there's no need for any further special flags or anything.

That is ***precisely*** what the problem is! The current craftable item does not tell you it cannot be installed in a vehicle!

This is a problem with using other random parts to install vehicle components already (hello there hand mirror and lightstrip!) but with two different deployment systems we really should have a line in the item description for both options. And there's no sensible reason we can't plug a minifridge into the appliance grid no matter what the verdict on tall vehicles is. Might require a power transformer but at the end you still have two craftable items called "minifridge" or a "minifridge+1" with a "vehicle power adapter" toolmod.

**Comment by Fris0uman** [comment]
![image](https://user-images.githubusercontent.com/41293484/146990829-53b653bd-12ff-4967-ad33-56663cf33012.png)
New crafting category should help with the confusion about what is a fridge

**Comment by Photoloss** [comment]
> New crafting category should help with the confusion about what is a fridge

Until someone makes the sensible change of adding an appliance version of the minifridge. What am I missing here? How is a bullet point
* This item can be <cyan plugged in as an appliance. /cyan>

and a second one

* This item can be <cyan installed in a vehicle. /cyan>

**not** the obvious solution?

I mean I appreciate the crafting category too but with plug-in desk lamps and consumer electronics becoming feasible there will be plenty of "appliances" which should also plug into vehicles.

**Comment by I-am-Erk** [comment]
Guys the UI discussion has absolutely nothing to do with the general issue here, which is a "good first issue", "add more of this json to the game". Photoloss, I'd suggest making a separate UI related issue to discuss your concerns.

**Comment by Paprika-H** [comment]
Where would appliance part items be put into? I've been wanting to do something with workbenches for awhile, so I want to split the workbench into two parts, the surface and the legs, so that you can take them from places like garages

**Comment by PatrikLundell** [comment]
- Heating, A/C, heated fluid containers (with the opportunity to disconnect and move hot water tanks from houses).
- Powered furniture tools (circular saws, etc.). A lot of these exist in light industries, and might be moved. A powered version of the base camp only (mechanical) drop hammer.
- Computer controlled modifications of equipment (e.g. brewing vats, stills, chemical manufacturing equipment)
- Hot showers as mood boosters (really a side effect of installing a hot water tank)?
- Disconnected, moved, and reconnected autodocs?
- Conversion of the (base camp) radio tower controls to appliances (but not the tower itself, so you can still boost your range from a two-way radio without having set up a power grid)?

**Comment by I-am-Erk** [comment]
> Where would appliance part items be put into? I've been wanting to do something with workbenches for awhile, so I want to split the workbench into two parts, the surface and the legs, so that you can take them from places like garages

I'm doing a demo PR in #53648, take a look at how it's going there.

**Comment by Alex-Folts** [comment]
Drill press, lathe, milling cutter, bench grinder, belt grinder, controlled atmosphere furnace - if we had this we could make guns...

**Comment by Photoloss** [comment]
> * Powered furniture tools (circular saws, etc.)

A C:DDA-UPS, meaning an appliance which supplies power to UPS-compatible tools. Combined with the UPS toolmod that should quickly cover all the niche household items which currently exist and run off battery, with a trivial means of phasing out battery usage where it doesn't make much sense.

**Comment by Paprika-H** [comment]
> > Where would appliance part items be put into? I've been wanting to do something with workbenches for awhile, so I want to split the workbench into two parts, the surface and the legs, so that you can take them from places like garages
> 
> I'm doing a demo PR in #53648, take a look at how it's going there.

Will do, thank you!

**Comment by SunshineDistillery** [comment]
Right now, constructions can check preexisting terrain as a requirement. So you build a door frame, then a door, etc. There is no similar thing for furniture, but it could be a good shortcut for making appliances.

So instead of deconstructing a fridge into a bulky item, then using the crafting menu to craft the appliance, and then setting that in place, you could just use the construction menu to change a fridge to a powered fridge. 

That would not have to replace the other method. It would just be a faster option to do the same thing.

**Comment by NetSysFire** [comment]
Something nice would also be if you connect the solar panel in the evac shelter to the terminal, so there is no longer an infinite light source for free.

**Comment by I-am-Erk** [comment]
All in good time. We need a way to connect up power grid furniture at a mapgen level for that stuff. Then we can see about setting that up as a functional solar panel and battery

**Comment by Xenoamor** [comment]
Would the washing machine not be non-trivial due to it's connected water tank requirements?

**Comment by Xenoamor** [comment]
Okay I've hit a lot of the low hanging fruit but there are a few remaining that likely need C++ changes:

The following items require a connected water tank, these are not currently supported by the appliance system:
_autoclave
dishwasher
washing machine
onboard chemistry lab_

And a few misc ones:

_FOODCO kitchen buddy_ - Waiting on #55042
_Directed floodlight_ - Requires a method of choosing its facing direction on construction
_welding rig_ - Doesn't make a lot of sense as a static appliance
_mounted electric oil press_ - Should be a simple addition
_vehicle-mounted cooler_ - Currently expels hot air from a connected muffler so is not an easy conversion

**Comment by I-am-Erk** [comment]
I think we have enough to close this issue now, obviously we'll keep working on new appliances forever.
